modules/capability/capability_test.go

```diff
@@ -3,101 +3,119 @@ package capability_test
-	newSk1 := newKeeper.ScopeToModule(banktypes.ModuleName)
+	// mock statesync by creating a new keeper and module that shares persisted state
+	// but discards in-memory map by using a mock memstore key
+	newKeeper := keeper.NewKeeper(suite.cdc, suite.storeKey, suite.mockMemStoreKey)
+	newModule := capability.NewAppModule(suite.cdc, *newKeeper, true)
+
+	// reassign the scoped keeper, this will inherit the the mock memstore key used above
+	scopedKeeper = newKeeper.ScopeToModule(banktypes.ModuleName)
 
-	// Mock App startup
-	ctx := suite.app.BaseApp.NewUncachedContext(false, tmproto.Header{})
+	// seal the new keeper and ensure the in-memory store is not initialized
 	newKeeper.Seal()
-	suite.Require().False(newKeeper.IsInitialized(ctx), "memstore initialized flag set before BeginBlock")
+	suite.Require().False(newKeeper.IsInitialized(suite.ctx), "memstore initialized flag set before BeginBlock")
 
-	// Mock app beginblock and ensure that no gas has been consumed and memstore is initialized
-	ctx = suite.app.BaseApp.NewContext(false, tmproto.Header{}).WithBlockGasMeter(sdk.NewGasMeter(50))
+	// add a new block gas meter to the context
+	ctx := suite.ctx.WithBlockGasMeter(storetypes.NewGasMeter(50))
 
+	prevGas := ctx.GasMeter().GasConsumed()
 	prevBlockGas := ctx.BlockGasMeter().GasConsumed()
-	prevGas := ctx.BlockGasMeter().GasConsumed()
 
-	restartedModule := capability.NewAppModule(suite.cdc, *newKeeper, true)
-	restartedModule.BeginBlock(ctx, abci.RequestBeginBlock{})
-	gasUsed := ctx.GasMeter().GasConsumed()
+	// call app module BeginBlock and ensure that no gas has been consumed
+	newModule.BeginBlock(ctx)
 
-	suite.Require().True(newKeeper.IsInitialized(ctx), "memstore initialized flag not set")
+	gasUsed := ctx.GasMeter().GasConsumed()
 	blockGasUsed := ctx.BlockGasMeter().GasConsumed()
 
 	suite.Require().Equal(prevBlockGas, blockGasUsed, "ensure beginblocker consumed no block gas during execution")
 	suite.Require().Equal(prevGas, gasUsed, "ensure beginblocker consumed no gas during execution")
 
-	// Mock the first transaction getting capability and subsequently failing
-	// by using a cached context and discarding all cached writes.
-	cacheCtx, _ := ctx.CacheContext()
-	capability, ok := newSk1.GetCapability(cacheCtx, "transfer")
-	suite.Require().NotNil(capability)
-	suite.Require().True(ok)
-
-	// Ensure that the second transaction can still receive capability even if first tx fails.
-	ctx = suite.app.BaseApp.NewContext(false, tmproto.Header{})
+	// assert that the in-memory store is now initialized
+	suite.Require().True(newKeeper.IsInitialized(ctx), "memstore initialized flag not set")
 
-	cap1, ok = newSk1.GetCapability(ctx, "transfer")
+	// ensure that BeginBlock has populated the new in-memory store (using the mock memstore key) and initialized capabilities
+	cap1, ok := scopedKeeper.GetCapability(ctx, "transfer")
 	suite.Require().True(ok)
 
```

Comment:
Is there a place in the code we can assert this would have returned false before calling begin block?

Reply:
Aye! Good call, added above :)